src/sdk/search/state.ts

```diff
@@ -0,0 +1,13 @@
+import { navigate } from 'gatsby'
```

Comment:
Why can't those functions be inside the framework? Also, the `applySearchState` can receive the navigation callback.

Take a look at this suggestion:
```ts
const applySearchState = (url: URL, navigate: (url: string) => void) => { ... }
```

Reply:
It's of like this already. The `SearchProvider` accepts the `onChange` callback. This callback returns a URL object instead of the string url.